src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ArrayInitHandler.java

```diff
@@ -74,7 +75,16 @@ protected DetailAST getLeftCurly() {
 
     @Override
     protected IndentLevel curlyIndent() {
-        final IndentLevel level = new IndentLevel(getIndent(), getBraceAdjustment());
+        int offset = 0;
+
+        final DetailAST lcurly = getLeftCurly();
+
+        if (isOnStartOfLine(lcurly)
+            && !TokenUtil.isOfType(lcurly.getParent(), TokenTypes.ARRAY_INIT)) {
+            offset = getBraceAdjustment();
```

Comment:
`curlyIndent` is returning the expected indentation for the left and right curlies.

If the parent of ARRAY_INIT isn't an ARRAY_INIT (which is almost a non-brace, then how can we use brace adjustment here which is only to be used for braces. Shouldn't `basicOffset` or `lineWrappingIndentation` be used here instead?

Reply:
@rnveach, sorry for late reply, I got disconnected with my own code.
 
It is for the case when lcurly is on new line, then curlies should be indented on `braceAdjustment` value ahead of the base.
It is for old style java codes.
```
class A 
  { // braceAdj
      void main() {} // braceAdj + basicOffset
  }  
```
```
class A {
    void main() {} //  basicOffset
}  
```

doc: 
![image](https://user-images.githubusercontent.com/48255244/102935857-f6d36b80-44cc-11eb-8c7a-d1154f24b297.png)